test/zipkin/util.go

```diff
@@ -107,5 +130,17 @@ func CheckZipkinPortAvailability() error {
 		return err
 	}
 	server.Close()
+
 	return nil
 }
+
+// LockAndReadZipkinStatusFlag RLocks and returns zipkinTracingEnabled flag
+func LockAndReadZipkinStatusFlag() bool {
```

Comment:
why not just lock and unlock here? Having them separate is error prone
```
func GetZipkinStatusFlag() {
  rlock
  defer runlock
  return value
}
```

Reply:
This would not work right because the read lock needs to be held until the client/method using the flag has completed retrieving zipkin trace. So for e.g. SpoofingClient reads the flag and sees that it is `true` and goes ahead and makes a Http Request. While the request is being satisfied some other routine can come can clean up the port-forwarding process because the Rlock was released and it was able to obtain a write lock. The read call on the Spoofingclient would fail as a result.

Reply:
@vagababov I Hope I answered your question. Please let me know if I have missed anything.

Reply:
Thanks, if we are reverting to per-suite as per your earlier suggestion sync.Once makes more sense. I have updated the changeset with Sync.Once approach.